app/views/devise/confirmations/new.html.erb

```diff
@@ -2,8 +2,14 @@
 <% proper_email = params[:email] || (resource.pending_reconfirmation? ? resource.unconfirmed_email : resource.email) %>
 <div class="crayons-layout crayons-layout--limited-l align-center mt-9 mb-10 p-3">
 
-  <h1 class="fs-5xl mb-5">Welcome to <%= community_name %>! 🎉</h1>
-  <h3 class="mb-5">A confirmation email has been sent to <a href="mailto:<%= proper_email %>"><%= proper_email %></a></h3>
+  <% if request.referer&.include?("/users/sign_up?state=email_signup") %>
```

Comment:
Maybe I'm being overly cautious here, but if the referrer has other query params that happen to come before `state`, such as `https://example.com/users/sign_up?i=i&state=email_signup` (since we use `i=i` in a lot of places), it would mean the same thing but not match that string.

To do it "right" (adaptable to change) would probably be more complex than we should put in a view template (parse the referer as a `URI`, check the URI path and `state` query param), but might be simple enough for a helper method.

Reply:
That is a fair concern. This if statement is there to conditionally hide the welcome message ("Welcome to Forem") and I could technically disregard the query param and only check for `/users/sign_up`. Would that make this more acceptable?

Reply:
Maybe you could do:

```suggestion
<% if request.referer&.include?(new_user_registration_path) %>
```

I would still consider moving this logic to https://github.com/forem/forem/blob/main/app/helpers/authentication_helper.rb so the `if` invokes a method that expresses intention in a clearer manner

Reply:
Updated!